### ISSUE TITLE
Having interface with the same name as stdlib type results in incorrect type in runtime

### ISSUE THREAD

**Issue by trilis** [Kotlin/dukat#381]
Consider library `mfiles`:
```
declare namespace MFiles {
	const Number: { new(): INumber; };
	function GetStringResource(resource: number): string;
}
interface INumber {
	Value: number;
	Clone(): INumber;
}
```
In generated code, in package `MFiles` we will see this:
```
external object Number {
}
external fun GetStringResource(resource: Number): String
```
Which will compile, but type of argument in `GetStringResource` will not be regular `kotlin.Number`, as intended by library author, but some irrelevant `MFiles.Number`.

**Comment by trilis** [comment]
Same problem with `jest/v16`:
```
declare namespace jasmine {
    interface Any {
        new (expectedClass: any): any;
        jasmineMatches(other: any): boolean;
        jasmineToString(): string;
    }
    function any(aclass: any): Any;
}
```

**Comment by JSMonk** [comment]
Fixed in #467
